ext/ruby_prof/rp_profile.c

```diff
@@ -586,6 +586,8 @@ static VALUE prof_initialize(int argc, VALUE* argv, VALUE self)
         prof_exclude_common_methods(self);
     }
 
+    rb_iv_set(cProfile, "@current", self);
```

Comment:
This create a reference from an instance of a profile to itself in Ruby. So let's not do that.

Instead, look at how the running? method is implemented and copy that.

Reply:
I'm pretty sure sets a class-level instance variable on `RubyProf::Profile` which points to the instance. Otherwise the test wouldn't work. I want to make the `RubyProf::Profile` instance available, not the `prof_profile_t*`.

Reply:
Right, sorry, I misread `cProfile` as `profile`.

However, I don't love this change because it is creating a global variable for a single use case. I can't see any reason why someone would create two different profiles at the same time, but it should work correctly, and this change would break it.

Having said that, I can live with the more limited change of the ruby-prof script setting it because that's a more limited change.

Reply:
We could just set it on `RubyProf::Cmd` instead.